src/cargo/ops/fix.rs

```diff
@@ -252,6 +252,59 @@ fn check_version_control(gctx: &GlobalContext, opts: &FixOptions) -> CargoResult
 }
 
 fn migrate_manifests(ws: &Workspace<'_>, pkgs: &[&Package]) -> CargoResult<()> {
+    // HACK: Duplicate workspace migration logic between virtual manifests and real manifests to
+    // reduce duplicate messages being reported to the user
+    if matches!(ws.root_maybe(), MaybePackage::Virtual(_)) {
+        // Warning: workspace's do not have an edition so this should only include changes needed by
```

Comment:
```suggestion
        // Warning: workspaces do not have an edition so this should only include changes needed by
```

I guess you mean this?